### DISCUSSION TITLE
LWT: an After Story

### DISCUSSION THREAD

**Discussion by HugoFara** [HugoFara/lwt#86] [Announcements]
# Post-LWT Era

In a previous post (#36), I stated that LWT 2.5 may be the last minor release of LWT, as I wanted to move to a new project codenamed "LW2". It was 2 months from now, and I made up my mind about future developments.

## Short Version (for those who don't have time)

In brief, I am considering to work with @jzohrab on his [LUTE](https://github.com/jzohrab/lute) project. Yet, as I'm no longer the only developer, it finally gives me some time to continue contributing to LWT. LWT may receive features from Lute on the future, as well as updates as long as they make the lives of users easier but I do not wish to diverge from the original LWT concept.

## The "LW2" Project Architecture Clarifies Itself

In brief, I now intend to work on a rebuild of LWT with more modern tools and frameworks. I recently started to collaborate with @jzohrab on his [Lute](https://github.com/jzohrab/lute) project, an effort to rebuild LWT with modern tools, and I think it is very close to what LW2 can be. So please try out the project and give us meaningful reviews! Even if I end up working on a different project, his code can still be an asset.

## Collaboration LWT/Lute

That being said, LWT has its own advantages: low level of abstraction, code tested by hundreds of users, etc. As @jzohrab is focusing only on Lute, it leaves me more time than expected and I can continue polishing LWT as useful features may be ported between Lute and LWT.

## Remaining Steps for LWT

As agreed with [lang-learn-guy](https://sourceforge.net/u/lang-learn-guy/profile/), a project that does not follow LWT concepts should not be named LWT. Which means for futures versions of LWT I now allow myself to publish major and minor releases a long as they don't diverge from the original LWT philosophy. As I see things, I want to focus on making the lives of users easier, and I don't plan on making big overhauls to the project.

It has been quite fun to refactor LWT this past year, so I'm gradually moving out of the project but still allows myself to contribute from times to times. I'll continue to provide bug fixes and answer to issues until further notice. So please enjoy and share LWT and LUTE, and happy language learning!

**Comment by jzohrab** [comment]
Seconding what @HugoFara wrote here.  Lute feels like a good evolution of the codebase, even if it's not the final form that this project should take.  I'm using Lute for my Spanish, it's working very well for me.  I'm on the LWT discord, and happy to answer any questions about the project.  Cheers and thanks again to Hugo F for his work on LWT. jz

**Comment by parradam** [comment]
Hey both, would be interested in seeing if I can contribute in some way, as a long-time fan of LWT! :)

**Reply by HugoFara** [reply]
Hey! A super-easy way to support the project is to drop a star :star2: or to create a fork :smile: 

If you are really motivated, you can share the project on social media, post a review somewhere here or on the Discord! Finally, I regularly post latest developments and following issues on the Discord, so it's a nice place to go if you want to discuss and contribute!

**Reply by parradam** [reply]
Sounds good!  If you have any good first issues, I would be happy to partake in the process too. :)

I'll join the Discord now!
